### DISCUSSION TITLE
[0.3.0] - 2023-03-06

### DISCUSSION THREAD

**Discussion by SkalskiP** [roboflow/supervision#37] [Announcements]
## 🚀 Added

New methods in `sv.Detections` API:
- `from_transformers` - convert Object Detection 🤗 Transformer result into `sv.Detections`
- `from_detectron2` - convert Detectron2 result into `sv.Detections`
- `from_coco_annotations` - convert COCO annotation into `sv.Detections`
- `area` - dynamically calculated property storing bbox area 
- `with_nms` - initial implementation (only class agnostic) of  `sv.Detections` NMS 

## 🌱 Changed

- Make `sv.Detections.confidence` field `Optional`.

## 🏆 Contributors

* @SkalskiP

<hr /><em>This discussion was created from the release <a href='https://github.com/roboflow/supervision/releases/tag/0.3.0'>[0.3.0] - 2023-03-06</a>.</em>

**Comment by mohamedaliELfeky** [comment]
In the ```from_coco_annotations```

I get this error:

AttributeError: type object 'Detections' has no attribute 'from_coco_annotations'

**Reply by onuralpszr** [reply]
Hello @mohamedaliELfeky first off, If you are using 0.3.0 version please update to latest first.  Secondly "from_coco_annotations" has been renamed and move into dataset section and for using loading coco 

https://github.com/roboflow/supervision/blob/develop/supervision/dataset/core.py

 ```python
            >>> import roboflow
            >>> from roboflow import Roboflow
            >>> import supervision as sv

            >>> roboflow.login()

            >>> rf = Roboflow()

            >>> project = rf.workspace(WORKSPACE_ID).project(PROJECT_ID)
            >>> dataset = project.version(PROJECT_VERSION).download("coco")

            >>> ds = sv.DetectionDataset.from_coco(
            ...     images_directory_path=f"{dataset.location}/train",
            ...     annotations_path=f"{dataset.location}/train/_annotations.coco.json",
            ... )

            >>> ds.classes
            ['dog', 'person']
```

**Reply by onuralpszr** [reply]
We should be also extra careful for release notes 

 https://github.com/roboflow/supervision/releases/tag/0.11.0   no deprecation or removal warning message in "release notes" 
 
 Sorry for confiusion

**Comment by Gitgutgait** [comment]
> Hello @mohamedaliELfeky first off, If you are using 0.3.0 version please update to latest first. Secondly "from_coco_annotations" has been renamed and move into dataset section and for using loading coco
> 
> https://github.com/roboflow/supervision/blob/develop/supervision/dataset/core.py
> 
> ```python
>            >>> import roboflow
>            >>> from roboflow import Roboflow
>            >>> import supervision as sv
> 
>            >>> roboflow.login()
> 
>            >>> rf = Roboflow()
> 
>            >>> project = rf.workspace(WORKSPACE_ID).project(PROJECT_ID)
>            >>> dataset = project.version(PROJECT_VERSION).download("coco")
> 
>            >>> ds = sv.DetectionDataset.from_coco(
>            ...     images_directory_path=f"{dataset.location}/train",
>            ...     annotations_path=f"{dataset.location}/train/_annotations.coco.json",
>            ... )
> 
>            >>> ds.classes
>            ['dog', 'person']
> ```

But if I change it like this :

detections = sv.DetectionDataset.from_coco(images_directory_path=TRAIN_DIRECTORY, 
                                           annotations_path=f"{TRAIN_DIRECTORY}/annotations.json")

Then i will get error for the labels:
# we will use id2label function for training
categories = TRAIN_DATASET.coco.cats
id2label = {k: v['name'] for k,v in categories.items()}
labels = [
    f"{id2label[class_id]}"
    **for _, _, class_id, _**
    in detections
]

ValueError: not enough values to unpack (expected 4, got 3)

**Reply by SkalskiP** [reply]
Hi 👋🏻 

First of all, change this:

```python
categories = TRAIN_DATASET.coco.cats
```

Into this:

```python
categories  = dataset.classes
```

Second of all, change this:

```python
labels = [
    f"{id2label[class_id]}"
    for _, _, class_id, _
    in detections
]
```

Into this:

```python
labels = [
    f"{categories[class_id]}"
    for _, _, _, class_id, _
    in detections
]
```

**Reply by hebrotem** [reply]
@SkalskiP the error message as more to do with unhashable type: 'numpy.ndarray'

![image](https://github.com/roboflow/supervision/assets/34936778/6acbc402-2be2-4646-8959-791e375886e0)

**Reply by hebrotem** [reply]
this might not be an efficient way, but it helped me:

![image](https://github.com/roboflow/supervision/assets/34936778/b1115f37-f53f-49ac-80ce-f0b440287284)

**Reply by Gitgutgait** [reply]
yes, it plots all the images in ds, but the original purpose was to plot the random selected image.

When using from_coco like this :
detections = sv.DetectionDataset.from_coco(images_directory_path = TRAIN_DIRECTORY,
                                           annotations_path=f"{TRAIN_DIRECTORY}/annotations.json")
 it take all the images and annotations from the directory

@SkalskiP, btw it didn't work, now it says :
ValueError: not enough values to unpack (expected 5, got 3)

**Reply by Gitgutgait** [reply]
If the last 4 statements is outside of the loop then it works as it should!
![image](https://github.com/roboflow/supervision/assets/60431155/b51b9d37-34ea-4109-a38d-cf586f640107)

**Reply by hebrotem** [reply]
I doubt if trying that would make it work as it should.

**Reply by Gitgutgait** [reply]
you were right, it didn't work as it should, but this is the updated code:
![image](https://github.com/roboflow/supervision/assets/60431155/b7dba264-39ab-42b0-b35f-9a02f4e86b07)
find the image path you want in the detections, and then use that one :D

**Comment by alandevkota** [comment]
I solved similar errors with this code:

![image](https://github.com/roboflow/supervision/assets/28220649/85b4dce7-d48b-44d0-b52b-215bd582fc7f)
